pandas/tseries/index.py

```diff
@@ -1637,6 +1637,24 @@ def delete(self, loc):
             new_dates = tslib.tz_convert(new_dates, 'UTC', self.tz)
         return DatetimeIndex(new_dates, name=self.name, freq=freq, tz=self.tz)
 
+    def putmask(self, mask, value):
+        """
+        return a new Index of the values set with the mask
+
+        See also
+        --------
+        numpy.ndarray.putmask
+        """
+        zzone = tslib.get_timezone(self.tzinfo)
```

Comment:
e.g. just like Period, I think we need something that asserts these are compatible boxed values

e.g.

```
def assert_compat_scalar(self, value):
    # do the tz comparison thing & return

    raise (Type or ValueError)
```

e.g. we do this kind of comparison in ``.equals/.insert`` as well